### ISSUE TITLE
Refine no-lonely-if rule

### ISSUE THREAD

**Issue by edg2s** [eslint/eslint#7562]
While the examples given on http://eslint.org/docs/rules/no-lonely-if appear to make sense, the rule forces a 4-option decision tree such as:
```
if ( x ) {
  if ( y ) {
    // x && y
  } else {
    // x & !y
  }
} else {
  if ( y ) {
    // !x && y
  } else {
    // !x & !y
  }
}
```

to be rewritten as:

```
if ( x ) {
  if ( y ) {
    // x && y
  } else {
    // x & !y
  }
} else if ( y ) {
  // !x && y
} else {
  // !x & !y
}
```

...which some may prefer, but is not obviously better than the more balanced version above.

It would be nice if we could enable this rule to only run if the nested `if` doesn't have an `else` itself, i.e. just:
```
if (foo) {
    // ...
} else {
    if (bar) {
        // ...
    }
}
```

This is currently a blocker to us including it in the Wikimedia style guide: https://github.com/wikimedia/eslint-config-wikimedia/issues/8

**Comment by eslintbot** [comment]
Hi @edg2s, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by edg2s** [comment]
I have provided all the information requested in 'proposing a rule change'. What information do you think is missing?

**Comment by not-an-aardvark** [comment]
Could you answer the questions in the [rule change proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md)?

**Comment by platinumazure** [comment]
@edg2s Even if all of the information is there, we do have to review many of these proposals per day and it helps us a lot when people take the time to structure their proposals according to our template. Thanks in advance!

**Comment by edg2s** [comment]
**What rule do you want to change?**
no-lonely-if

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option

**Please provide some example code that this change will affect:**

```js
if ( x ) {
  if ( y ) {
    // x && y
  } else {
    // x & !y
  }
} else {
  if ( y ) {
    // !x && y
  } else {
    // !x & !y
  }
}
```

**What does the rule currently do for this code?**
errors due to the if nested inside an else

**What will the rule do after it's changed?**
not error when an extra option is passed to only fail when the nested if has not else branches.

**Comment by not-an-aardvark** [comment]
Thanks for filling out the template.

Personally, I'm not a big fan of this proposal. It seems like it would encourage a style like this:

```js
if (foo) {
  bar();
else {
  if (baz) {
    qux();
  } else {
    if (bar) {
      foo();
    } else {
    }
  }
}
```

In my opinion, this is messy and defeats the point of the `no-lonely-if` rule. However, let's see what the rest of the team thinks.

**Comment by edg2s** [comment]
Well it /allows/ that style rather than encourages it. The point of the relaxing option would be to allow the user to write easy to read and balanced nested if branches, whereas the current rules requires it to be either unbalanced or completely flat.

To my mind the main point of the rule is to fix *simple* else{if(){ blocks.

**Comment by kaicataldo** [comment]
Closing this since there hasn't been any discussion in quite a while and progress on the issue has stalled. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. 

Feel free to open a new issue with a proposal to continue this discussion if this is something you would like to continue advocating for. Please also remember that you can always create this rule for yourself and distribute it as a plugin.
